firewood/src/proof.rs

```diff
@@ -86,7 +85,7 @@ pub struct ProofNode {
     /// Otherwise, the node's value or the hash of its value.
     pub value_digest: Option<ValueDigest<Box<[u8]>>>,
     /// The hash of each child, or None if the child does not exist.
-    pub child_hashes: [Option<HashType>; BranchNode::MAX_CHILDREN],
+    pub child_hashes: [Option<HashType>; BranchConstants::MAX_CHILDREN],
```

Comment:
Pro tip: make a separate PR that pulls out these BranchConstants.

I don't think they should be separated from BranchNode without good cause though.